torch/_inductor/comms.py

```diff
@@ -226,9 +228,14 @@ def decide_global_ordering_of_comms(nodes: List[BaseSchedulerNode]):
     TODO: Come up with a better approach
     """
     comm_nodes = [n for n in nodes if is_collective(n.node)]
+
+    def item(x: Set[str]) -> str:
+        assert len(x) == 1
+        return next(iter(x))
+
     for i in range(1, len(comm_nodes)):
         # Enforce ordering by making previous comm a `WeakDep` dependency of the next comm
-        comm_nodes[i].add_fake_dep(WeakDep(comm_nodes[i - 1].get_name()))
+        comm_nodes[i].add_fake_dep(WeakDep(item(comm_nodes[i - 1].get_buffer_names())))
```

Comment:
@peterbell10 I believe this might be causing an error in multi-gpu runs: https://productionresultssa6.blob.core.windows.net/actions-results/fd465c4e-dbd7-42de-bbe3-a1aadc133a4b/workflow-job-run-958b70a4-e428-5161-9e96-1ac739b382a7/logs/job/job-logs.txt?rsct=text%2Fplain&se=2024-07-22T20%3A35%3A12Z&sig=AMvNE0N7lR5HFJsAqcB0GFtgXnlavcD1Ny6WPf7w9Kc%3D&ske=2024-07-23T05%3A54%3A19Z&skoid=ca7593d4-ee42-46cd-af88-8b886a2f84eb&sks=b&skt=2024-07-22T17%3A54%3A19Z&sktid=398a6654-997b-47e9-b12b-9515b896b4de&skv=2023-11-03&sp=r&spr=https&sr=b&st=2024-07-22T20%3A25%3A07Z&sv=2023-11-03  cc. @atalman @yifuwang

Reply:
Your link requires some kind of authorization, can you share the error?

Reply:
I believe this is the error:
```
2024-07-22T14:29:20.7393861Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_inductor/graph.py", line 1638, in codegen
2024-07-22T14:29:20.7394041Z     self.scheduler = Scheduler(self.operations)
2024-07-22T14:29:20.7394662Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_dynamo/utils.py", line 246, in time_wrapper
2024-07-22T14:29:20.7394786Z     r = func(*args, **kwargs)
2024-07-22T14:29:20.7395433Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_inductor/scheduler.py", line 1551, in __init__
2024-07-22T14:29:20.7395638Z     comms.decide_global_ordering_of_comms(self.nodes)
2024-07-22T14:29:20.7396393Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_inductor/comms.py", line 238, in decide_global_ordering_of_comms
2024-07-22T14:29:20.7396802Z     comm_nodes[i].add_fake_dep(WeakDep(item(comm_nodes[i - 1].get_buffer_names())))
2024-07-22T14:29:20.7397376Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_inductor/comms.py", line 233, in item
2024-07-22T14:29:20.7397534Z     assert len(x) == 1
2024-07-22T14:29:20.7397895Z torch._dynamo.exc.BackendCompilerFailed: backend='inductor' raised:
2024-07-22T14:29:20.7398013Z AssertionError: 
2024-07-22T14:29:20.7398019Z 
2024-07-22T14:29:20.7398319Z Set TORCH_LOGS="+dynamo" and TORCHDYNAMO_VERBOSE=1 for more information
2024-07-22T14:29:20.7398333Z 
2024-07-22T14:29:20.7398338Z 
2024-07-22T14:29:20.7398627Z You can suppress this exception and fall back to eager by setting:
2024-07-22T14:29:20.7398753Z     import torch._dynamo
2024-07-22T14:29:20.7398940Z     torch._dynamo.config.suppress_errors = True
2024-07-22T14:29:20.7398950Z 
2024-07-22T14:29:20.7398955Z 
2024-07-22T14:29:20.7399219Z To execute this test, run the following from the base repo dir:
2024-07-22T14:29:20.7399972Z     python test/distributed/test_compute_comm_reordering.py -k TestComputeCommReorderingMultiProc.test_reorder_compute_for_overlap
```

This PR should make sure the unit test runs on regular PR CI: https://github.com/pytorch/pytorch/pull/131415, it will be cool to have this PR stack on top of that PR for easier testing. Thanks!

Reply:
I've added ciflow/periodic in the reverted PRs